ompi/mca/osc/rdma/osc_rdma_accumulate.c

```diff
@@ -117,16 +128,47 @@ static inline int ompi_osc_rdma_cas_local (const void *source_addr, const void *
                                            uint64_t target_address, mca_btl_base_registration_handle_t *target_handle,
                                            ompi_osc_rdma_module_t *module, bool lock_acquired)
 {
+    int ret;
+    void *compare_copy;
+    void *result_copy;
+    bool compare_copied = false;
+    bool result_copied = false;
+
     OSC_RDMA_VERBOSE(MCA_BASE_VERBOSE_TRACE, "performing compare-and-swap with local regions");
 
-    memcpy (result_addr, (void *) (uintptr_t) target_address, datatype->super.size);
+    osc_rdma_accelerator_mem_copy(result_addr, (void *) (uintptr_t) target_address, datatype->super.size);
+    /* Check if either addr is accelerator. Will have to perform some copies if so */
+    if (osc_rdma_is_accel(result_addr)) {
```

Comment:
We already checked whether `result_addr` is a device addr in `osc_rdma_accelerator_mem_copy`, not we're checking again. Can we save this repeated check?

Reply:
I'll think about it